python/ray/tune/experiment/trial.py

```diff
@@ -954,6 +956,13 @@ def __setstate__(self, state):
         if not self.stub:
             validate_trainable(self.trainable_name)
 
+        # Update the resource specification
```

Comment:
Here's how I'm updating the resource config: recompute the `placement_group_factory` from the trainable instead of just loading it from the saved trial state.

I'm including it in this PR so that I don't need to log warnings just to remove them immediately with a follow-up fix. This will handle: (1) overwriting with an AIR trainer with a different `ScalingConfig`, and (2) overwriting trainable using `with_resources` with different resource specification.

Added a test for (1) to `train/tests/test_tune.py`.